src/assets/Generator.Shared/RequestHeaderExtensions.cs

```diff
@@ -64,6 +64,16 @@ public static void Add(this RequestHeaders headers, string prefix, IDictionary<s
             }
         }
 
+        public static void Add(this RequestHeaders headers, string name, ETag? value)
```

Comment:
You can add extension methods for `RequestConditions` and `MatchConditions` as well, since the code that adds those headers will be the same for all cases:
```cs
public static void Add(this RequestHeaders headers, string name, ETag value)
{
    headers.Add(name, value.ToString());
}

public static void Add(this RequestHeaders headers, MatchConditions conditions)
{
    if (conditions.IfMatch != null)
    {
        headers.Add("If-Match", conditions.IfMatch.Value);
    }

    if (conditions.IfNoneMatch != null)
    {
        headers.Add("If-None-Match", conditions.IfNoneMatch.Value);
    }
}

public static void Add(this RequestHeaders headers, RequestConditions conditions, string format)
{
    if (conditions.IfMatch != null)
    {
        headers.Add("If-Match", conditions.IfMatch.Value);
    }

    if (conditions.IfNoneMatch != null)
    {
        headers.Add("If-None-Match", conditions.IfNoneMatch.Value);
    }

    if (conditions.IfModifiedSince != null)
    {
        headers.Add("If-Modified-Since", conditions.IfModifiedSince.Value, format);
    }

    if (conditions.IfUnmodifiedSince != null)
    {
        headers.Add("If-Unmodified-Since", conditions.IfUnmodifiedSince.Value, format);
    }
}
```